lib/rack/utils.rb

```diff
@@ -447,7 +447,7 @@ def [](k)
       end
 
       def []=(k, v)
-        canonical = k.downcase.freeze
```

Comment:
the intention here is to avoid mutations, frozen string literal pragma doesn't really do the same thing

```
# frozen_string_literal: true

canonical = 'test'.downcase
puts canonical.frozen?
=> false
```

Reply:
Thank you for the comment.
My mistake. I was misunderstanding the behaviour of frozen string literal pragma.
As you mentioned, frozen string literal pragma does not freeze the var `canonical` in this case.
Reverted the modification referred by your comment.

Thanks.